akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1284,11 +1281,8 @@ private[stream] object Collect {
               if (!hasBeenPulled(in)) tryPull(in)
               push(out, elem)
             } else buffer.enqueue(elem)
-          case other ⇒
-            val ex = other match {
-              case Failure(t)              ⇒ t
-              case Success(s) if s == null ⇒ ReactiveStreamsCompliance.elementMustNotBeNullException
-            }
+          case Success(null) ⇒
```

Comment:
pull here?
```
if (!hasBeenPulled(in)) tryPull(in)
```

Reply:
Thanks for spotting it! Actually I think
```
if (isClosed(in) && todo == 0) completeStage()
else if (!hasBeenPulled(in)) tryPull(in)
``` 
would be correct. Will add testcase to cover stage completion when null is emitted last.